intermediate_notebooks/examples/custreamz/parsing_haproxy_logs.ipynb

```diff
@@ -0,0 +1,427 @@
+    "    syslog_timestamp = clean_df[1].data.cat([clean_df[2].data, clean_df[3].data, clean_df[4].data], sep=' ')\n",
+    "    clean_df['syslog_timestamp'] = cudf.Series(syslog_timestamp)\n",
+    "    for col in [1,2,3,4]:\n",
+    "        clean_df.drop_column(col)\n",
+    "\n",
+    "    program_pid_df = clean_df[5].str.split('[')\n",
+    "    program_sr = program_pid_df[0]\n",
+    "    pid_sr = program_pid_df[1]\n",
+    "    clean_df['program'] = program_sr\n",
+    "    clean_df['pid'] = pid_sr.str.rstrip(']:')\n",
+    "    clean_df = clean_df.drop(labels=[5])\n",
+    "    del program_pid_df\n",
+    "\n",
+    "    client_pid_port_df = clean_df[6].str.split(':')\n",
+    "    clean_df['client_ip'], clean_df['client_port'] = client_pid_port_df[0], client_pid_port_df[1]\n",
+    "    clean_df.drop_column(6)\n",
+    "    del client_pid_port_df\n",
+    "\n",
+    "    clean_df['accept_date'] = clean_df[7].str.lstrip('[').str.rstrip(']')\n",
+    "    clean_df.drop_column(7)\n",
+    "\n",
+    "    clean_df.rename({8: 'frontend_name'}, inplace=True)\n",
+    "    backend_server_df = clean_df[9].str.split('/')\n",
+    "    clean_df['backend_name'], clean_df['server_name'] = backend_server_df[0], backend_server_df[1]\n",
+    "    clean_df.drop_column(9)\n",
+    "\n",
+    "    time_cols = ['time_request', 'time_queue', 'time_backend_connect', 'time_backend_response', 'time_duration']\n",
+    "    time_df = clean_df[10].str.split('/')\n",
+    "    for col_id, col_name in enumerate(time_cols):\n",
+    "        clean_df[col_name] = time_df[col_id]\n",
+    "    clean_df.drop_column(10)\n",
+    "    del time_df\n",
+    "\n",
+    "    clean_df.rename({11: 'http_status_code'}, inplace=True)\n",
+    "    clean_df.rename({12: 'bytes_read'}, inplace=True)\n",
+    "    clean_df.rename({13: 'captured_request', 14: 'captured_response', 15: 'termination_state'}, inplace=True)\n",
+    "\n",
+    "    con_cols = ['actconn', 'feconn', 'beconn', 'srvconn', 'retries']\n",
+    "    con_df = clean_df[16].str.split('/')\n",
+    "    for col_id, col_name in enumerate(con_cols):\n",
+    "        clean_df[col_name] = con_df[col_id]\n",
+    "    clean_df.drop_column(16)\n",
+    "    del con_df\n",
+    "\n",
+    "    q_df = clean_df[17].str.split('/')\n",
+    "    clean_df['srv_queue'], clean_df['backend_queue'] = q_df[0], q_df[1]\n",
+    "    clean_df.drop_column(17)\n",
+    "    del q_df\n",
+    "    \n",
+    "    clean_df['time_backend_response'] = clean_df['time_backend_response'].astype('int')\n",
+    "    clean_df['http_status_code'] = clean_df['http_status_code'].astype('int')\n",
+    "    \n",
+    "    # Return the cuDF dataframe so that other operations in the stream can use it\n",
+    "    return clean_df\n",
+    "    \n",
+    "# We now use Streamz to create a Stream from Kafka by polling the topic every 1s. If you changed Dask=True, please ensure you have a Dask cluster up and running\n",
```

Comment:
Why not go ahead and create the dask cuda cluster for the user and use it?

Is there a benefit to using single-GPU streaming mode (without Dask)? If so, adding some comments explaining that would be helpful.

If not, this repository has examples using cuStreamz with single-GPU cuDF already.